src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    @Test
+    void userDataCorrect_ok() {
+        registerService.register(testUser);
+        boolean actual = Storage.people.contains(testUser);
+        assertTrue(actual);
+    }
+
+    @Test
+    void userAreNull_notOk() {
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(null);
+        }, "User can not be null");
+    }
+
+    @Test
+    void userAgeAreLessThat18_notOk() {
+        testUser.setAge(INVALID_AGE);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User age must be 18 or bigger");
+    }
+
+    @Test
+    void userAgeAreNegative_notOk() {
+        testUser.setAge(NEGATIVE_AGE);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User age must be bigger then 0");
+    }
+
+    @Test
+    void userAgeAreBiggerThen100_notOk() {
+        testUser.setAge(BIG_AGE);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User age must be less then 100");
+    }
+
+    @Test
+    void userAgeAreNull_notOk() {
+        testUser.setAge(null);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User age must be not null");
+    }
+
+    @Test
+    void userPasswordAreNull_notOk() {
+        testUser.setPassword(null);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User password must be not null");
+    }
+
+    @Test
+    void userPasswordAreInvalid_notOk() {
+        testUser.setPassword(INVALID_PASSWORD);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "User password must be 6 symbol or bigger");
+    }
+
+    @Test
+    void userLoginAreNull_notOk() {
+        testUser.setLogin(null);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "Login can`t be null");
+    }
+
+    @Test
+    void userLoginInvalid_notOk() {
+        testUser.setLogin(INVALID_LOGIN);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "Login must contain @gmail.com");
+    }
+
+    @Test
+    void userIdAreNull_notOk() {
+        testUser.setId(null);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "Id can`t be null");
+    }
+
+    @Test
+    void userIdAreNegative_notOk() {
+        testUser.setId(NEGATIVE_ID);
+        assertThrows(IncorrectDataExeption.class, () -> {
+            registerService.register(testUser);
+        }, "Id can`t be less then 0");
+    }
+
+    @Test
+    void storageAlreadyContainsUser_notOk() {
+        registerService.register(testUser);
```

Comment:
You can’t use method register() here because you testing this method. You need to add a user to the base like this: Storage.people.add().
And then register the user with the same email using register() method expecting an exception